src/util.rs

```diff
@@ -12,3 +12,44 @@ pub fn is_audio_file(url: &CowStr) -> bool {
     }
     false
 }
+
+// return the extension of an url as a string
+pub fn get_ext(url: &CowStr) -> String {
+    let path = Path::new(url.as_ref());
+    if let Some(ext_osstr) = path.extension() {
+        ext_osstr.to_string_lossy().to_lowercase()
+    } else {
+        String::new()
+    }
+}
+
+// return mimetype given an extension
+pub fn get_mimetype(ext: &str) -> String {
```

Comment:
likewise, I think this should return a &str or Cow rather than allocating a String that might not be needed by the caller

Reply:
it's not about memory leak, but rather best practice is to avoid allocations for performance, but I also believe in doing benchmarks and letting performance optimizations come from actual slow spots